### PR TITLE
Kernel: Bail out earlier from Process::lookup_stacks_directory

### PR DESCRIPTION
**Pull request by Hendiadyoin1**
Just a small PR, to remove an 8 months old FIXME.

This also condenses the code a bit

### CODE REVIEW

File: Kernel/ProcessSpecificExposed.cpp [hunk 1]
```diff
@@ -61,19 +61,19 @@ ErrorOr<NonnullRefPtr<Inode>> Process::lookup_stacks_directory(const ProcFS& pro
 {
     ErrorOr<NonnullRefPtr<ProcFSProcessPropertyInode>> thread_stack_inode { ENOENT };
 
-    // FIXME: Try to exit the loop earlier
+    auto maybe_needle = name.to_int();
+    if (!maybe_needle.has_value())
+        return ENOENT;
+    pid_t needle = maybe_needle.release_value();
```
Comment (IdanHo) [inline Kernel/ProcessSpecificExposed.cpp:67]: `auto`

**Review by IdanHo — COMMENTED** [review]
Looks great, just one nitpick

File: Kernel/ProcessSpecificExposed.cpp [hunk 2]
```diff
@@ -59,21 +59,20 @@ ErrorOr<void> Process::traverse_stacks_directory(FileSystemID fsid, Function<Err
 
 ErrorOr<NonnullRefPtr<Inode>> Process::lookup_stacks_directory(const ProcFS& procfs, StringView name) const
 {
-    ErrorOr<NonnullRefPtr<ProcFSProcessPropertyInode>> thread_stack_inode { ENOENT };
+    auto maybe_needle = name.to_int();
```
[the remark this answers is not in the document]
Comment (Hendiadyoin1) [reply Kernel/ProcessSpecificExposed.cpp:62]: That causes clang to complain about sign compare

[the remark this answers is not in the document]
Comment (IdanHo) [reply Kernel/ProcessSpecificExposed.cpp:62]: I'm not sure what you mean.

[the remark this answers is not in the document]
Comment (Hendiadyoin1) [reply Kernel/ProcessSpecificExposed.cpp:62]: with that needle and tid would have different signage, which we warn and error about when comparing them
```
../../Kernel/ProcessSpecificExposed.cpp:71:20: error: comparison of integers of different signs: 'unsigned int' and 'int' [-Werror,-Wsign-compare]
        if (needle == tid) {
```

[the remark this answers is not in the document]
Comment (IdanHo) [reply Kernel/ProcessSpecificExposed.cpp:62]: The previous line verifies that the signed value is non-negative, and all signed positive values fit in an unsigned int, I don't see why you can't just cast it.

**Pull request merged**